datafusion/expr/src/logical_plan/builder.rs

```diff
@@ -121,6 +123,39 @@ impl LogicalPlanBuilder {
         }))
     }
 
+    /// A named temporary relation with a schema.
+    ///
+    /// This is used to represent a relation that does not exist at the
```

Comment:
perhaps it is worth mentioning that this is to implement CTEs?

Reply:
this should be clear now via https://github.com/matthewgapp/arrow-datafusion/blob/460561cc47004e02e47b9b4aea4703c021de58dc/datafusion/expr/src/logical_plan/builder.rs#L127